### ISSUE TITLE
Incorrect assertion for PartialContent test

### ISSUE THREAD

**Issue by bgerstle** [8thlight/cob_spec#72]
According to RFC 7233 section 2.1: "if the [suffix length] value is greater than or equal to the the length...the byte range is interpreted as the remainder of the representation." In other words, if I request the range `1-1000` for a file that's 5 bytes long, I should get a valid response with `Content-Range: 1-4`. 

Instead, cob_spec requests the range `75-80` for a file that's 77 bytes long, and expects to receive a response with the `416` status code and a `Content-Range` header with the value `*/77`.

**Comment by bgerstle** [comment]
Example failure: 
![image](https://user-images.githubusercontent.com/444217/46041369-35a0ce00-c0d8-11e8-824a-4a3f40741d63.png)

**Comment by bgerstle** [comment]
Here are some examples of invalid ranges that could be used for the test:

- `10-0` (last > first)
- `-78` (suffix greater than length of file)
- `-0` (empty suffix range)
- `78-` (first byte position beyond end of file)

**Comment by trptcolin** [comment]
👍 makes sense to me. Any objections @jdesrosiers or others?

**Comment by jdesrosiers** [comment]
@bgerstle, Good catch! Can you create a PR for this? We should keep and fix the test for this scenario, then add others for the 416 cases.

Ideally it would be great if we could break the PartialContent test into multiple tests and separate them into their own suite.
